src/Autofac/Core/Registration/ComponentRegistryBuilder.cs

```diff
@@ -61,6 +58,15 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
+        /// <inheritdoc />
+        protected override ValueTask DisposeAsync(bool disposing)
+        {
+            _registeredServicesTracker.Registered -= OnRegistered;
+            _registeredServicesTracker.RegistrationSourceAdded -= OnRegistrationSourceAdded;
+
+            return _registeredServicesTracker.DisposeAsync();
+        }
```

Comment:
This comment is on the other spots where base is intentionally not called.

```suggestion

            // Do not call the base, otherwise the standard Dispose will fire.
        }
```